tox.ini

```diff
@@ -104,4 +114,7 @@ whitelist_externals =
 commands =
     pip install -e .
 	git clone --branch {env:BRANCH:master} git@github.com:gridsingularity/gsy-backend-integration-tests.git {envtmpdir}/gsy-backend-integration-tests
-	bash -c 'for f in {envtmpdir}/gsy-backend-integration-tests/gsy_e_tests/integration_tests/*.feature; do behave --tags=-disabled "$f" || exit 1; done'
+	bash -c 'cp -r {envtmpdir}/gsy-backend-integration-tests/gsy_e_tests/integration_tests/ ./integration_tests/'
+	bash -c 'for f in ./integration_tests/*.feature; do behave --tags=-disabled "$f" || exit 1; done'
+commands_post =
+	bash -c '[ -z "{env:INTEGRATION_TESTS_REPO_URL:}" ] && rm -rf ./integration_tests/ || :'
```

Comment:
This will make the use of `envtmpdir` pointless AFAIS. Also, where do you set the `INTEGRATION_TESTS_REPO_URL` env variable for local tox runs?

Reply:
I understand why you say it.
However, if we were to let here the previous approach and try to fix the import of the module (which I tried), more import references will have also to be updated, because the `envtmpdir` lives in a rout in which the import does not work as it used to. For example, after fixing the import of the function on integration_tests.py module, then it ran with the following Exception related to the export path:

```
Traceback (most recent call last):
File "/Users/GSy/Workspace/gsy-e/.tox/integrationtests/lib/python3.8/site-packages/behave/model.py", line 1329, in run
match.run(runner.context)
File "/Users/GSy/Workspace/gsy-e/.tox/integrationtests/lib/python3.8/site-packages/behave/matchers.py", line 98, in run
self.func(context, *args, **kwargs)
File ".tox/integrationtests/tmp/gsy-backend-integration-tests/gsy_e_tests/integration_tests/steps/integration_tests.py", line 681, in run_sim
raise Exception(er)


 Exception: [Errno 2] No such file or directory: '/Users/GSy/Workspace/gsy-e/gsy_e-simulation/integration_tests/two_sided_market.infinite_bus/2022-01-31T15:49:29/raw_data/2022-01-31T03:00.json'
```

And even though the files were created, they would not be read. 

If  `behave ./integration_tests/` is used on the testenv scenario instead of `bash -c 'for f in ./integration_tests/*.feature; do behave --tags=-disabled "$f" || exit 1; done'` , all of the tests are executed and you can see more than one fail do to similar reasons. For this to be "fixed" differently, it would be required to update multiple files and path references. On the first approach I came up with (the open discussed offline on a meeting) I had done this in a way but can get tricky.

IMO this might appear to be un-neat, but it is indeed more straight forward. Also, with this command, the cleanup is only done when run locally, and as it is "inside" a bash command, it does not need to whitelist anything else apart from `bash`.

---------

The  `INTEGRATION_TESTS_REPO_URL` env variable was set by Marcelo on the github workflow. If this variable is passed, then the execution would be online. Otherwise, it is local. `[ -z "{env:INTEGRATION_TESTS_REPO_URL:}" ]` returns True if the variable is not set (defaults to empty), and if True, `rm -rf ./integration_tests/` will then be executed. Otherwise, nothing happens.

Reply:
> The INTEGRATION_TESTS_REPO_URL env variable was set by Marcelo on the github workflow. If this variable is passed, then the execution would be online. Otherwise, it is local

Sorry, I still didn't understand this part :) What does it mean that the execution would be "online" vs "local"? In both cases we would still be cloning the same repository, right?

Reply:
In this case, by "online" we understand the tests being executed by a Github workflow.
By "local", we understand running them directly on the cli by means of `tox -e integrationtests` or `tox -e ci`.

And yes, in both cases we would clone the same repository, but if cloned "online", as `gsy-backend-integration-tests repo` is private, it is cloned by using an https url, with some credentials attached to it. 
On the other hand, if executed locally, the clone can be done by means of a ssh url. Trying to clone "online" with the ssh url resuted in a _no right permissions_ error.

Regarding removing `./integration_tests` if the tests are run locally, it is to leave the local repository clean of that folder which is only temporarily needed to execute the tests correctly. If ran "online", AFAIU, it would be in a "virtual machine", which would be reset after the workflow execution...

Reply:
Could we clone in both cases using HTTPS, so that we don't have to define two variables and two flows? In gsy-web we have [this line](https://github.com/gridsingularity/gsy-e-sdk/blob/dfa138a90703e9376b92c801fa878b6f3de26ad5/integration_tests/build_test_containers.sh#L11) that uses it and I think it works well.

Regarding the deletion of `./integration_tests`: I would personally just delete it also during the github action. I think that it's good to keep the same execution flow in all our environments, as much as possible. And also because this would simplify the code without slowing it down (I suppose that the deletion is very fast).

Reply:
> Could we clone in both cases using HTTPS, so that we don't have to define two variables and two flows? In gsy-web we have [this line](https://github.com/gridsingularity/gsy-e-sdk/blob/dfa138a90703e9376b92c801fa878b6f3de26ad5/integration_tests/build_test_containers.sh#L11) that uses it and I think it works well.

No, because with HTTPS you need to first login to github and with the SSH approach you can provide a key that we can pass through the github action env.
As Eduardo correctly mentions, for public repos, this is all not a problem and we could indeed use only HTTPS, but for private ones, we need to authenticate.